driver/cam_hal.c

```diff
@@ -374,7 +374,11 @@ esp_err_t cam_config(const camera_config_t *config, framesize_t frame_size, uint
     cam_obj->height = resolution[frame_size].height;
 
     if(cam_obj->jpeg_mode){
+#ifdef CAMERA_JPEG_MODE_FRAME_SIZE_AUTO
```

Comment:
Should be `CONFIG_ CAMERA_JPEG_MODE_FRAME_SIZE_AUTO`. You missed the `CONFIG_` prefix